src/Database/DBStructure.php

```diff
@@ -737,8 +739,10 @@ public static function update($basePath, $verbose, $action, $install = false, ar
 		self::checkInitialValues();
 
 		if ($action && !$install) {
-			DI::config()->set('system', 'maintenance', 0);
-			DI::config()->set('system', 'maintenance_reason', '');
+			if (!$in_maintenance) {
```

Comment:
This condition should be the opposite.

Reply:
No. We only have to reset the maintenance mode when it isn't already in.

Reply:
The parameters are now clarified.